web/src/project/ProjectSummary.tsx

```diff
@@ -0,0 +1,169 @@
+/* eslint react-hooks/exhaustive-deps: 0 */
+
+import * as React from 'react'
+import { useSearchParams } from 'react-router-dom'
+import _ from 'lodash'
+
+import { AppContext } from '../GlobalState'
+import { ProjectSelector } from './ProjectSelector'
+import { WebApi, ProjectSummaryResponse } from '../sm-api/api'
+
+import { Input, Button } from 'reactstrap'
+
+const PAGE_SIZES = [20, 40, 100, 1000]
+
+
+export const ProjectSummary = () => {
+
+    // store project in global settings
+    const globalContext = React.useContext(AppContext)
+
+    let [searchParams, setSearchParams] = useSearchParams()
+
+    const directToken = searchParams.get('token')
+
+    const [summary, setSummary] = React.useState<ProjectSummaryResponse | undefined>()
+    const [suppliedDirectToken, setSuppliedDirectToken] = React.useState<boolean | undefined>(undefined)
+    const [pageTokens, setPageTokens] = React.useState<number[]>([])
+    const [isLoading, setIsLoading] = React.useState<boolean>(false)
+    const [error, setError] = React.useState<string | undefined>()
+    const [pageLimit, _setPageLimit] = React.useState<number>(PAGE_SIZES[0])
+
+    async function getProjectSummary(token: any, addToPageToken: boolean) {
```

Comment:
If you define a function within a component that uses state/context variables from the outer scope, the closure could become stale when React runs a state update and you might end up with strange bugs. For this particular use case `useCallback` is a hook provided by the `react` package such that your function will be re-defined if any dependencies change - in this case `[setIsLoading, globalContext, pageLimit, pageTokens, setPageTokens]`.  I'm not sure if there will be any bugs in this particular situation, but just a heads-up!